HumanControls/HumanControls.cpp

```diff
@@ -0,0 +1,137 @@
+#include "HumanControls.h"
+
+const char *HumanControls::disabled = "disabled";
+const char *HumanControls::enabled = "enabled";
+const char *HumanControls::primed = "primed";
+const char *HumanControls::status = HumanControls::disabled;
+const char *HumanControls::lastStatus = HumanControls::disabled;
+
+HumanControls::HumanControls(unsigned int encoderPinClk,
+                             unsigned int encoderPinDt,
+                             unsigned int displayAddress,
+                             unsigned int displayLen,
+                             unsigned int displayWidth,
+                             unsigned int angleIncrement,
+                             unsigned int angleMin,
+                             unsigned int angleMax,
+                             unsigned int pressureIncrement,
+                             unsigned int pressureMin,
+                             unsigned int pressureMax,
+                             unsigned int durationIncrement,
+                             unsigned int durationMin,
+                             unsigned int durationMax,
+                             unsigned int numButtons,
+                             unsigned int encoderPinSW,
+                             unsigned int enablePin,
+                             unsigned int firePin)
+    : m_menuController(encoderPinClk, encoderPinDt, displayAddress, displayLen, displayWidth,
+                       angleIncrement, angleMin, angleMax, pressureIncrement, pressureMin,
+                       pressureMax, durationIncrement, durationMin, durationMax),
+      m_pinDebouncer(numButtons), m_enableController(), m_fireController() //,
+                                                                           /*encoderPinSW(encoderPinSW), enablePin(enablePin), firePin(firePin)*/
+{
+    this->m_encoderPinSW = encoderPinSW;
+    this->m_enablePin = enablePin;
+    this->m_firePin = firePin;
+    this->m_isConnected = false;
+}
+
+void HumanControls::init()
+{
+    connect();
+    m_pinDebouncer.addPin(this->m_encoderPinSW, HIGH, INPUT_PULLUP);
+    m_pinDebouncer.addPin(this->m_enablePin, HIGH, INPUT_PULLUP);
+    m_pinDebouncer.addPin(this->m_firePin, HIGH, INPUT_PULLUP);
+    m_pinDebouncer.begin();
+    m_menuController.init(status);
+}
+
+void HumanControls::update()
+{
+    if (m_isConnected == false)
+    {
+        this->connect();
+    }
+
+    this->setStatus();
+
+    m_menuController.menuUpdate(status);
+    m_pinDebouncer.update();
+}
+
+void HumanControls::setStatus()
```

Comment:
No action needed here, but eventually this won't be necessary. The robot's status will be in the JsonState object. That and `m_isConnected` will be all that's needed.